tests/testthat/test-measures.R

```diff
@@ -1,64 +1,275 @@
-test_that("riem_measures returns the right output", {
+#######################
+# test required params
+#######################
+
+test_that("riem_measures fails if required 'station' param is absent", {
+  expect_snapshot_error(riem_measures(date_start = "2014-03-01"))
+})
+
+test_that("riem_measures fails if required 'date_start' param is absent", {
+  expect_snapshot_error(riem_measures(station = "VOHY"))
+})
+
+
+###################
+# param validation
+###################
+
+test_that("riem_measures validates 'station' param", {
+  # wrong type
+  expect_snapshot_error(riem_measures(station = 11111L,
+                                      date_start = "2014-03-01"))
+  # invalid value
+  expect_snapshot_error(riem_measures(station = "ZZZZZ",
+                                      date_start = "2014-03-01"))
+})
+
+test_that("riem_measures validates dates", {
+  # date_start is invalid value
+  expect_snapshot_error(riem_measures(station = "VOHY",
+                                      date_start = "somethingelse"))
+  # date_start is badly formatted
+  expect_snapshot_error(riem_measures(station = "VOHY",
+                                      date_start = "2015 31 01"))
+
+  # date_end is invalid value
+  expect_snapshot_error(riem_measures(station = "VOHY",
+                                      date_start = "2014-03-01",
+                                      date_end = "somethingelse"))
+  # date_end is badly formatted
+  expect_snapshot_error(riem_measures(station = "VOHY",
+                                      date_start = "2014-03-01",
+                                      date_end = "2015 31 01"))
+
+  # date_end is before date_start
+  expect_snapshot_error(riem_measures(station = "VOHY",
+                                      date_start = "2015-12-01",
+                                      date_end = "2013-12-01"))
+})
+
+# TODO validate 'data' against [all, tmpf, dwpf, relh, drct, sknt, p01i, alti, mslp, vsby, gust, skyc1, skyc2, skyc3, skyc4, skyl1, skyl2, skyl3, skyl4, wxcodes, ice_accretion_1hr, ice_accretion_3hr, ice_accretion_6hr, peak_wind_gust, peak_wind_drct, peak_wind_time, feel, metar, snowdepth] # nolint: line_length_linter
```

Comment:
should we tackle this in this PR?

Reply:
Short answer:  I don't think so.

Longer answer:  I left this TODO here as a placeholder while I ponder the merits of such a check. The more I think about it, the less I'm convinced of its value. `data` is the closest thing we have to a pass-through parameter. Since the only value our code counts in is `all`, and since we already test that case with a snapshot, I'm not sure what additional benefit that adding this check would provide. Certainly it would increase our dependence on IEM's data structure, as this test would need updated if IEM ever added, renamed, or removed any fields. Unless you object, I'm in favor to remove the TODO and instead rely on the existing snapshot to test what we care about.